api/src/config.rs

```diff
@@ -0,0 +1,81 @@
+//! All the configurations for the application
+//!
+//! Contains types definitions and traits implementations
+//! for managed states used with `.manage()` on a rocket app`
+
+use diesel::mysql::MysqlConnection;
+use r2d2;
+use r2d2_diesel::ConnectionManager;
+use rocket::http::RawStr;
+use rocket::http::Status;
+use rocket::request::{self, FromParam, FromRequest};
+use rocket::{Outcome, Request, State};
+use std::ops::Deref;
+
+/// Encapsulate a r2d2 Mysql connection pool.
+type Pool = r2d2::Pool<ConnectionManager<MysqlConnection>>;
+
+/// Connection request guard type: a wrapper around an r2d2 pooled connection.
+pub struct DbConn(pub r2d2::PooledConnection<ConnectionManager<MysqlConnection>>);
+
+/// A string representing admin token to access admin only endpoints
+#[derive(Clone, PartialEq)]
+pub struct AdminToken(pub String);
+
+/// Contains all the application configuration objects
+pub struct MyConfig {
+    /// The database pool
+    pub pool: Pool,
+    /// The admin token
+    pub admin_token: AdminToken,
+}
+
+/// Initializes a database pool.
+pub fn init_pool(database_url: &str) -> Pool {
+    let manager = ConnectionManager::<MysqlConnection>::new(database_url);
+    r2d2::Pool::new(manager).expect("error creating database pool")
```

Comment:
Error format is `error: <details>`, preferably starting with a verb.

Eg: `error: can't create database pool`

Reply:
done